src/navigation/screenTracking.js

```diff
@@ -0,0 +1,43 @@
+import analytics from '@react-native-firebase/analytics';
+
+export const gaTrackScreen = (route) => {
+  if (!route?.name) return;
+
+  let name;
+
+  const titleCase = (str) => {
+    if (typeof str !== 'string') return '';
+    return str.replace(/([A-Z])/g, ' $1').replace(/^./, function (str) {
+      return str.toUpperCase();
+    });
+  };
+
+  // For classifiers use the project slug. For project list add the type of project list. For anything else convert to title case.
+  switch (route.name) {
+    case 'QuestionClassifier':
+      name = route?.params?.project?.slug ?? route.name;
+      break;
+    case 'SwipeClassifier':
+      name = route?.params?.project?.slug ?? route.name;
+      break;
+    case 'DrawingClassifier':
+      name = route?.params?.project?.slug ?? route.name;
+      break;
+    case 'MultiAnswerClassifier':
+      name = route?.params?.project?.slug ?? route.name;
+      break;
+    case 'ProjectList':
+      name = route?.params?.selectedProjectTag
+        ? `Project List - ${titleCase(route.params.selectedProjectTag)}`
+        : 'Project List';
+      break;
+    // As a default convert the route name to title case.
+    default:
+      name = titleCase(route.name);
+  }
+
+  analytics().logScreenView({
```

Comment:
I don't think it's really necessary, but wrapping this (or maybe `gaTrackScreen(newRoute)` in RootNavigator) in a try/catch crossed my mind. For consideration.

Reply:
Thanks, it might not be needed but if it could fail if the code changes and something other than an object is passed to the function, so I think it's a good idea to add the try/catch so those errors get logged. I've updated the code.